### PR TITLE
Organize Imports in packages avoiding tree printer

### PR DESCRIPTION
**Pull request by wpopielarski**
Work funded on #144. PR thought as a military drill area to collect opinions about tree printer replacement.

Right now Group Imports has been implemented: 	Adds Group Imports functionality (2d67fc2)

### CODE REVIEW

**Comment by sschaef** [comment]
Please rebase onto master

File: src/main/scala/scala/tools/refactoring/implementations/OrganizeImports.scala [hunk 1]
```diff
@@ -12,6 +12,7 @@ import common.Change
 import common.TreeTraverser
```
Comment (mlangc) [inline src/main/scala/scala/tools/refactoring/implementations/OrganizeImports.scala]: Just a general comment regarding the commit message: "Innards" are what you can see here (https://www.google.at/search?q=innards&client=firefox-b-ab&source=lnms&tbm=isch&sa=X&ved=0ahUKEwj0t7K5i-TNAhWBRhQKHdpkA-4Q_AUICCgB&biw=1920&bih=1032)... you mean internals I guess ;-)

File: src/test/scala/scala/tools/refactoring/tests/implementations/imports/OrganizeImportsTest.scala [hunk 2]
```diff
@@ -4296,4 +4296,98 @@ class OrganizeImportsTest extends OrganizeImportsBaseTest {
...
+    import testing.Messages.Implicits._
+    import testing.acme.Implicits.application
+
+    class Tested {
+      implicit val defaultLang: String = "IT"
+      def foo = Messages(
+    }
+    """ isNotModified
+  } applyRefactoring organizeCustomized(dependencies = Dependencies.RecomputeAndModify)
+  //play.api.i18n.Messages.apply("dskdlskd")(play.api.i18n.Messages.Implicits.applicationMessages(i18n.this.Lang.defaultLang, play.api.Play.current))
```
Comment (sschaef) [inline src/test/scala/scala/tools/refactoring/tests/implementations/imports/OrganizeImportsTest.scala]: I guess this comment shouldn't be here?

File: src/main/scala/scala/tools/refactoring/analysis/CompilationUnitDependencies.scala [hunk 3]
```diff
@@ -310,10 +329,10 @@ trait CompilationUnitDependencies extends CompilerApiExtensions with ScalaVersio
             case Import(select @ Select(Ident(nme.scala_), `language`), feature) =>
               feature foreach (selector => addToResult(Select(select, selector.name)))
 
-            case imp @ Import(qualifier, selector)  => {
+            case imp @ Import(qualifier, selector) => {
               if (inScopeForLocalImports) {
                 if (!qualifier.symbol.isLocal && !qualifier.symbol.isVal) {
-                  if (isRelativeToTopLevelImports(qualifier) || isRelativeToEnclosingPackage(qualifier)) {
+                  if (if (newWay) false else isRelativeToTopLevelImports(qualifier) || isRelativeToEnclosingPackage(qualifier)) {
```
Comment (sschaef) [inline src/main/scala/scala/tools/refactoring/analysis/CompilationUnitDependencies.scala]: This nested if is unnecessary.

**Comment by sschaef** [comment]
From my point of view, this LGTM. Any code clean-ups or better Scaladoc you want to do Wieslaw?

**Comment by wpopielarski** [comment]
S.
I fix this `t.attachments.isEmpty` tomorrow. I have to go now.

File: src/main/scala/scala/tools/refactoring/analysis/CompilationUnitDependencies.scala (hunk 3 above)
Comment (wpopielarski) [inline src/main/scala/scala/tools/refactoring/analysis/CompilationUnitDependencies.scala]: I think it is more readable than `!newWay && isRelativeToTopLevelImports(qualifier)`. `newWay` just shows me pieces of code for modification.

**Pull request merged**